### PR TITLE
Add documentation for z3py compatibility API

### PR DESCRIPTION
**Pull request by nafur**
This PR includes documentation for the z3py compatibility API into our general API.
It does so by adding the z3py compatibility API as an external project to download it and then have sphinx document it via the `autodoc` extension that we already use for our regular python API.
Right now we simply show everything on one page, which should be refactored in the future.

### CODE REVIEW

**Comment by alex-ozdemir** [comment]
Looks good to me, but someone else should take a look too.

**Review by alex-ozdemir — APPROVED** [review]

File: docs/api/python/python.rst [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-    datatypedecl
-    datatypeselector
-    grammar
-    op
-    result
-    roundingmode
-    solver
-    sort
-    unknownexplanation
+Most of the time, use the :doc:`z3py compatibility API <z3compat/z3compat>`.
```
Comment (yoni206) [inline docs/api/python/python.rst:25]: How about saying something more specific, and having arguments for both? For example:
```
If you are a new user, or already have an application that uses Z3's python API, use the :doc:`z3py compatibility API <z3compat/z3compat>`.

If you would like a more feature-complete python API, with the ability to do almost everything that the cpp API allows, use the :doc:`regular Python API <regular/python>`.
```

**Review by yoni206 — COMMENTED** [review]
LGTM, one comment.

File: docs/api/python/python.rst (hunk 1 above)
Comment (nafur) [reply docs/api/python/python.rst]: Thanks!

**Pull request merged**